Rakefile

```diff
@@ -38,4 +40,25 @@ namespace :beaker do
     args.with_defaults(:test_files => 'acceptance/performance/')
     run_beaker(args[:test_files])
   end
+
+  desc "Run beaker based acceptance tests, leaving VMs in place for later runs"
+  task :first_run, [:test_files] do |t, args|
+    args.with_defaults(:test_files => 'acceptance/tests/')
+    ENV["BEAKER_PRESERVE_HOSTS"] = "always"
+    run_beaker(args[:test_files])
+  end
+
+  desc "Re-run beaker based acceptance tests, using previously provisioned VMs"
+  task :rerun, [:test_files] do |t, args|
```

Comment:
Given the manual step required in between, I think it's better as two
tasks. Once that goes away, then I can automate away the difference and
this won't matter (I hope).

On Wed, Apr 1, 2015 at 3:46 PM, Andrew Roetker <notifications@github.com>
wrote:

> In Rakefile
> <https://github.com/puppetlabs/puppetdb/pull/1337#discussion_r27619544>:
>
> > @@ -38,4 +40,25 @@ namespace :beaker do
> >      args.with_defaults(:test_files => 'acceptance/performance/')
> >      run_beaker(args[:test_files])
> >    end
> > +
> > +  desc "Run beaker based acceptance tests, leaving VMs in place for later runs"
> > +  task :first_run, [:test_files] do |t, args|
> > +    args.with_defaults(:test_files => 'acceptance/tests/')
> > +    ENV["BEAKER_PRESERVE_HOSTS"] = "always"
> > +    run_beaker(args[:test_files])
> > +  end
> > +
> > +  desc "Re-run beaker based acceptance tests, using previously provisioned VMs"
> > +  task :rerun, [:test_files] do |t, args|
>
> Could these two things (first_run/rerun) be combined to :run or something
> with an arg for --test-only or --no-provisioning?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetdb/pull/1337/files#r27619544>.
>